shell/common/shorebird/blobs_handle.h

```diff
@@ -0,0 +1,47 @@
+#ifndef SHELL_COMMON_SHOREBIRD_BLOBS_HANDLE_H_
+#define SHELL_COMMON_SHOREBIRD_BLOBS_HANDLE_H_
+
+#include "flutter/fml/file.h"
+#include "flutter/runtime/dart_snapshot.h"
+#include "third_party/dart/runtime/include/dart_tools_api.h"
+
+namespace flutter {
+
+// An offset into an indexed collection of buffers. blob is the index of the
+// buffer, and offset is the offset into that buffer.
+struct BlobsIndex {
+  size_t blob;
+  size_t offset;
+};
+
+// Implements a POSIX file I/O interface which allows us to provide the four
+// data blobs of a Dart snapshot (vm_data, vm_instructions, isolate_data,
+// isolate_instructions) to Rust as though it were a single piece of memory.
+class SnapshotsDataHandle {
+ public:
+  // This would ideally be private, but we need to be able to call it from the
```

Comment:
You can list createForSnapshots as a friend method.

Reply:
Actually, it should already work, since createForSnapshots is a static on this class.

Reply:
https://stackoverflow.com/questions/65286678/make-unique-cannot-access-private-constructor-in-static-member